src/redux/flows/flows.types.ts

```diff
@@ -14,54 +19,44 @@ export type LoadFlowAction = {
 export type CreateNewFlowAction = {
   type: typeof CREATE_NEW_FLOW;
 };
-export type Flow = {
-  selected: {
-    type?: "link" | "node" | "port";
-    id?: string;
-  };
-  hovered: {
-    type?: "link" | "node" | "port";
-    id?: string;
-  };
-  scale: number;
-  nodes?: {
-    [nodeName: string]: {
-      id: string;
-      label: string;
-      ports: {
-        outPort: {
-          id: string;
-          type: string;
-        };
-        inPort?: {
-          id: string;
-          type: string;
-        };
-      };
-      properties: Object;
-      position: { x: number; y: number };
-      needs?: Object;
-      send_to?: Object;
-      depth?: number;
-    };
+
+export interface Node extends Omit<INode, "type"> {
+  label?: string;
+  needs?:
+    | {
+        encode: boolean;
```

Comment:
why does this interface specifically define `encode`?

Reply:
It shouldn't :)
Weird that it works.

I would change it to [pod: string]: boolean

Also now that I see it in 51 from pea to pod, which would be more accurate

Reply:
Changed it